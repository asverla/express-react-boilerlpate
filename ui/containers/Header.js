import React, { Component } from 'react'
import AppBar from 'material-ui/AppBar'
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem'
import {Link} from 'react-router-dom'

class Header extends Component {
  render () {
    return (
      <div id='header'>
        <AppBar id='appBar' title='Express React Boilerplate' />
        <Drawer id='drawer' open={this.props.isOpen}>
          <MenuItem><Link to='/about'>About</Link></MenuItem>
          <MenuItem><Link to='/demo'>Demo</Link></MenuItem>
        </Drawer>
      </div>
    )
  }
}
export default Header
