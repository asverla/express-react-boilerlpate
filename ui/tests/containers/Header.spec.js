import React from 'react'
import chai from 'chai'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from '../../containers/Header'
import AppBar from 'material-ui/AppBar'
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem/MenuItem'

Enzyme.configure({ adapter: new Adapter() })

describe('When Header component is rendered', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Header />)
  })
  test('it should render the AppBar', () => {
    chai.expect(wrapper.find(AppBar)).to.exist
  })

  test('it should render AppBar which has title \'Express React Boilerplate\'', (params) => {
    let title = 'Express React Boilerplate'
    chai.expect(wrapper.find(AppBar).props().title).to.equal(title)
  })

  test('it should render the Drawer', () => {
    chai.expect(wrapper.find(Drawer)).to.exist
  })

  test('it should render Drawer with two MenuItems', () => {
    chai.expect(wrapper.find(Drawer).children()).to.have.lengthOf(2)
    chai.expect(wrapper.find(Drawer).find(MenuItem)).to.have.lengthOf(2)
  })
  test('it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  afterEach(() => {
    wrapper.unmount()
  })
})
