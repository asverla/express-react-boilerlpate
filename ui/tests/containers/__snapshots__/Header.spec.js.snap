// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Header component is rendered it should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <b />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AppBar
          id="appBar"
          showMenuIconButton={true}
          title="Express React Boilerplate"
          zDepth={1}
        />,
        <Drawer
          disableSwipeToOpen={false}
          docked={true}
          id="drawer"
          open={null}
          openSecondary={false}
          swipeAreaWidth={30}
          width={null}
          zDepth={2}
        >
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
          >
            <Link
              replace={false}
              to="/about"
            >
              About
            </Link>
          </MenuItem>
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
          >
            <Link
              replace={false}
              to="/demo"
            >
              Demo
            </Link>
          </MenuItem>
        </Drawer>,
      ],
      "id": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "appBar",
          "showMenuIconButton": true,
          "title": "Express React Boilerplate",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            >
              <Link
                replace={false}
                to="/about"
              >
                About
              </Link>
            </MenuItem>,
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            >
              <Link
                replace={false}
                to="/demo"
              >
                Demo
              </Link>
            </MenuItem>,
          ],
          "disableSwipeToOpen": false,
          "docked": true,
          "id": "drawer",
          "open": null,
          "openSecondary": false,
          "swipeAreaWidth": 30,
          "width": null,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "children": <Link
                replace={false}
                to="/about"
              >
                About
              </Link>,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "About",
                "replace": false,
                "to": "/about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "children": <Link
                replace={false}
                to="/demo"
              >
                Demo
              </Link>,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Demo",
                "replace": false,
                "to": "/demo",
              },
              "ref": null,
              "rendered": "Demo",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppBar
            id="appBar"
            showMenuIconButton={true}
            title="Express React Boilerplate"
            zDepth={1}
          />,
          <Drawer
            disableSwipeToOpen={false}
            docked={true}
            id="drawer"
            open={null}
            openSecondary={false}
            swipeAreaWidth={30}
            width={null}
            zDepth={2}
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            >
              <Link
                replace={false}
                to="/about"
              >
                About
              </Link>
            </MenuItem>
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            >
              <Link
                replace={false}
                to="/demo"
              >
                Demo
              </Link>
            </MenuItem>
          </Drawer>,
        ],
        "id": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "appBar",
            "showMenuIconButton": true,
            "title": "Express React Boilerplate",
            "zDepth": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/about"
                >
                  About
                </Link>
              </MenuItem>,
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/demo"
                >
                  Demo
                </Link>
              </MenuItem>,
            ],
            "disableSwipeToOpen": false,
            "docked": true,
            "id": "drawer",
            "open": null,
            "openSecondary": false,
            "swipeAreaWidth": 30,
            "width": null,
            "zDepth": 2,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "children": <Link
                  replace={false}
                  to="/about"
                >
                  About
                </Link>,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                  "replace": false,
                  "to": "/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "children": <Link
                  replace={false}
                  to="/demo"
                >
                  Demo
                </Link>,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Demo",
                  "replace": false,
                  "to": "/demo",
                },
                "ref": null,
                "rendered": "Demo",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
